lib/howl/interactions/file_selection.moon

```diff
@@ -0,0 +1,302 @@
+-- Copyright 2012-2015 The Howl Developers
+-- License: MIT (see LICENSE.md at the top-level directory of the distribution)
+
+glib = require 'ljglibs.glib'
+
+import app, clipboard, config, interact, log, Project from howl
+import File from howl.io
+import markup, style, ListWidget from howl.ui
+import Matcher from howl.util
+
+append = table.insert
+separator = File.separator
+
+style.define_default 'directory', 'key'
+style.define_default 'filename', 'string'
+
+howl.config.define
+  name: 'hidden_file_extensions'
+  description: 'File extensions that determine which files should be hidden in file selection lists'
+  scope: 'global'
+  type_of: 'string_list'
+  default: {'a', 'bc', 'git', 'hg', 'o', 'pyc', 'so'}
+
+home_dir = -> File glib.get_home_dir!
+
+root_dir = (file) ->
+  if not file
+    return root_dir home_dir!
+  while file.parent
+    file = file.parent
+  return file
```

Comment:
General style comment: In the rest of the code implicit returns are used, except for early returns.